src/main/java/io/english/service/AssignmentService.java

```diff
@@ -0,0 +1,49 @@
+package io.english.service;
+
+import io.english.entity.dao.Assignment;
+import io.english.entity.dao.AssignmentItem;
+import io.english.entity.dao.AssignmentType;
+import io.english.entity.request.AssignmentRequest;
+import io.english.exceptions.EntityNotFoundException;
+import io.english.mappers.AssignmentItemMapper;
+import io.english.mappers.AssignmentMapper;
+import io.english.repository.AssignmentRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import java.time.LocalDateTime;
+import java.util.List;
+
+@Service
+@RequiredArgsConstructor
+public class AssignmentService {
+    private final AssignmentRepository assignmentRepository;
+    private final UserService userService;
+
+    public Assignment getById(Long id) {
+        return assignmentRepository.findById(id).orElseThrow(
+                () -> new EntityNotFoundException(String.format("Assignment not found by id=%d", id)));
+    }
+
+    public Assignment createAssignment(AssignmentRequest assignmentRequest) {
+        Assignment assignment = AssignmentMapper.INSTANCE.toEntity(assignmentRequest);
+        var user = userService.getCurrentUser();
```

Comment:
instead of getting user in createAssignmen request, user id should be the argument like 
`public Assignment createAssignment(AssignmentRequest assignmentRequest, Long createdById)`